base/array.jl

```diff
@@ -834,6 +834,41 @@ function setindex!(A::Array{T}, X::Array{T}, c::Colon) where T
     return A
 end
 
+"""
+    setindex(collection, value, key...)
+
+Create a new collection with the element/elements of the location specified by `key`
+replaced with `value`(s).
+
+# Implementation
+
+`setindex(collection, value, key...)` must have the property such that
+
+```julia
```

Comment:
Turn this into a `jldoctest`?

Reply:
It's a bit tricky but I can write something like this

`````
```jldoctest; output = false, setup = :(using Base: setindex; copy′(x) = x .+ 0.0; x = [1]; value = 0.5; key = (1,))
y1 = setindex(x, value, key...)

y2 = copy′(x)
@assert convert.(eltype(y2), x) == y2

y2[key...] = value
@assert convert.(eltype(y2), y1) == y2

# output

```
`````

and then `# output` will be hidden in the HTML output.  However, it would be still visible in the REPL (`?Base.setindex`) and I think it'd be a bit confusing.  I think not using jldoctest is maybe slightly better as both text and HTML outputs are concise this way.

Reply:
OK, that's fine.